diesel/src/mysql/types/mod.rs

```diff
@@ -5,39 +5,231 @@ mod date_and_time;
+            bytes.len() <= 2,
+            "Received more than 2 bytes decoding u16. \
+             Was a Integer expression accidentally identified as SmallInt?"
+        );
+        debug_assert!(
+            bytes.len() >= 2,
+            "Received fewer than 2 bytes decoding u16. \
+             Was the expression accidentally identified as SmallInt?"
+        );
+        bytes
+            .read_u16::<<Mysql as Backend>::ByteOrder>()
+            .map_err(Into::into)
+    }
+}
+
+impl ToSql<Unsigned<Integer>, Mysql> for u32 {
+    fn to_sql<W: Write>(&self, out: &mut Output<W, Mysql>) -> serialize::Result {
+        out.write_u32::<<Mysql as Backend>::ByteOrder>(*self)
+            .map(|_| IsNull::No)
+            .map_err(Into::into)
+    }
+}
+
+impl FromSql<Unsigned<Integer>, Mysql> for u32 {
+    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
+        let mut bytes = not_none!(bytes);
+        debug_assert!(
+            bytes.len() <= 4,
+            "Received more than 4 bytes decoding u32. \
+             Was a BigInteger expression accidentally identified as Integer?"
+        );
+        debug_assert!(
+            bytes.len() >= 4,
+            "Received fewer than 4 bytes decoding u32. \
+             Was a SmallInteger expression accidentally identified as Integer?"
+        );
+        bytes
+            .read_u32::<<Mysql as Backend>::ByteOrder>()
+            .map_err(Into::into)
+    }
+}
+
+impl ToSql<Unsigned<Integer>, Mysql> for u64 {
+    fn to_sql<W: Write>(&self, out: &mut Output<W, Mysql>) -> serialize::Result {
+        out.write_u64::<<Mysql as Backend>::ByteOrder>(*self)
+            .map(|_| IsNull::No)
+            .map_err(Into::into)
+    }
+}
+
+impl FromSql<Unsigned<BigInt>, Mysql> for u64 {
+    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
+        let mut bytes = not_none!(bytes);
+        debug_assert!(
+            bytes.len() <= 8,
+            "Received more than 8 bytes decoding u64. \
+             Was the expression accidentally identified as BigInt?"
+        );
+        debug_assert!(
+            bytes.len() >= 8,
+            "Received fewer than 8 bytes decoding u64. \
+             Was a Integer expression accidentally identified as BigInt?"
+        );
+        bytes
+            .read_u64::<<Mysql as Backend>::ByteOrder>()
+            .map_err(Into::into)
+    }
+}
+
+impl ToSql<Bool, Mysql> for bool {
     fn to_sql<W: Write>(&self, out: &mut Output<W, Mysql>) -> serialize::Result {
         let int_value = if *self { 1 } else { 0 };
-        <i32 as ToSql<sql_types::Integer, Mysql>>::to_sql(&int_value, out)
+        <i32 as ToSql<Integer, Mysql>>::to_sql(&int_value, out)
     }
 }
 
-impl FromSql<sql_types::Bool, Mysql> for bool {
+impl FromSql<Bool, Mysql> for bool {
     fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
         Ok(not_none!(bytes).iter().any(|x| *x != 0))
     }
 }
 
+impl<ST> HasSqlType<Unsigned<ST>> for Mysql
```

Comment:
I'm not sure if this is a good idea, because this will allow also things like `Unsigned<Bool>` to be valid in some contexts. (Note for @diesel-rs/reviewers)

Reply:
@weiznich would it be better to have a `HasSqlType` for `SmallInt`, `Integer`, and `BigInt`, instead of this more generic version?

Reply:
I don't think the `HasSqlType` impl being generic is a problem. It doesn't enable anything on its own.